xml_converter/src/attribute/euler_rotation.cpp

```diff
@@ -2,9 +2,35 @@
 
 #include <string>
 #include <vector>
+#include <typeinfo>
 
 #include "../rapid_helpers.hpp"
 #include "../rapidxml-1.13/rapidxml.hpp"
-#include "../string_helper.hpp"
 
-using namespace std;
+EulerRotation parse_euler_rotation(rapidxml::xml_attribute<>* input, vector<XMLError*> *){
+    EulerRotation euler_rotation;
+    vector<string> compound_values;
+    string attributename; 
+    attributename = get_attribute_name(input); 
+    compound_values = split(get_attribute_value(input), ",");
+    
+    if (typeid(compound_values) == typeid(std::string)) {
+        if (attributename == "rotatex") {	
+            euler_rotation.x_rotation = std::stof(get_attribute_value(input)); 
+    	}
+        else if (attributename == "rotatey") {
+            euler_rotation.y_rotation = std::stof(get_attribute_value(input)); 
+        }
+        else if (attributename == "rotatez") {
+            euler_rotation.z_rotation = std::stof(get_attribute_value(input)); 
+        }
+    else {    
+        euler_rotation.x_rotation = std::stof(compound_values[0]);    
+        euler_rotation.y_rotation = std::stof(compound_values[1]);    
+        euler_rotation.z_rotation = std::stof(compound_values[2]); 
+        }
+    }
+
+    return euler_rotation;
```

Comment:
This euler_rotation may have partially uninitialized fields.

Reply:
Understood. Compound values and multi flag values now have default values.